opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1362,73 +1362,117 @@ template cpgrid::Entity<3> Dune::createEntity(const CpGrid&, int, bool);
-    //
-    // Build the LGR/level1 from the selected patch of cells from level0 (level0 = this->data_[0]).
-    const auto& [level1_ptr, boundary_old_to_new_corners, boundary_old_to_new_faces, parent_to_children_faces,
-                 parent_to_children_cells, child_to_parent_faces, child_to_parent_cells]
-        = (*(this-> data_[0])).refinePatch(cells_per_dim, startIJK, endIJK);
-    // Add level 1 to "data".
-    (this-> data_).push_back(level1_ptr);
-    //
-    // LEVEL 0, definition/declaration of some members:
-    (*data_[0]).level_data_ptr_ = &(this -> data_);
-    (*data_[0]).level_ = 0;
-    // Relation between level and leafview cell indices.
-    std::map<int,int>& l0_to_leaf_cells = (*data_[0]).level_to_leaf_cells_;
-    // For level0, attach children to each parent cell. EMPTY entry for no parents.
+    // Total amount of patches:
+    const int& num_patches = startIJK_vec.size();
+    assert(cells_per_dim_vec.size() == startIJK_vec.size());
+    assert(cells_per_dim_vec.size() == endIJK_vec.size());
+    std::vector<int> all_patch_corners; 
+    std::vector<int> all_patch_faces; 
+    std::vector<int> all_patch_cells;
+    std::vector<std::vector<std::array<int,2>>> all_patch_boundary_old_to_new_corners;
+    std::vector<std::vector<std::tuple<int,std::vector<int>>>> all_patch_boundary_old_to_new_faces;
+    std::vector<std::vector<std::tuple<int,std::vector<int>>>> all_patch_parent_to_children_cells;
+    std::vector<std::vector<std::array<int,2>>> all_patch_child_to_parent_cells;
+    all_patch_boundary_old_to_new_corners.reserve(num_patches);
+    all_patch_boundary_old_to_new_faces.reserve(num_patches);
+    all_patch_parent_to_children_cells.resize(num_patches);
+    all_patch_child_to_parent_cells.resize(num_patches);
+    // For level0, attach children to each parent cell. For no parents, entry {-1, {-1}} representing {no level, {no children}}
     auto& l0_parent_to_children_cells = (*data_[0]).parent_to_children_cells_;
-    const std::vector<int>& no_child = {-1};
-    const int& no_level = -1;
-    const std::tuple<int, std::vector<int>>& no_children = std::make_tuple(no_level, no_child);
-    // For cells with no children, we set {-1,{-1}}. Entries of actual parents will be rewritten.
     l0_parent_to_children_cells.resize(data_[0]-> size(0));
-    for (int cell = 0; cell < data_[0] -> size(0); ++cell){
-        l0_parent_to_children_cells[cell] = no_children;
-    }
-    // For level1/LGR, attach to each child-cell its parent.
-    std::vector<std::array<int,2>>& l1_child_to_parent_cells = (*data_[1]).child_to_parent_cells_;
-    // For cells with no parent, we set {-1,-1}. Entries with actual parents will be rewritten.
-    const std::array<int,2>& no_parent = {-1,-1};
-    l1_child_to_parent_cells.resize(data_[1]-> size(0));
-    for (int cell = 0; cell < data_[1]->size(0); ++cell){
-        l1_child_to_parent_cells[cell] = no_parent;
+    const std::vector<int> no_children = {-1};    
```

Comment:
Why isn't this an empty vector (representing no children)?

Reply:
l1_child_to_parent_cells is not empty in the end since it contains for each cell of the LGR (level 1, coming from refining a patch from level 0),  l1_child_to_parent_cells[child cell index] = {0, parent cell index} (0 representing level 0). The deleted lines are not needed since this container gets populated later with an actual parent for each entry. In contrary to leaf_child_to_parent_cells which may have entries with no parent, those are "by defualt set as" {-1, -1} representing "no level" and "no parent"

Reply:
I meant <any-level-from-1-to-last-lgr>.child_to_parent_cells_ behave as I described above (each cell has an actual parent in level 0, so each entry looks like {0, parent cell index}), and <leafview>.child_to_parent_cells_ may have entries {-1,-1} (instead of empty entries) when the cell does not have a parent.